crates/pathfinder/src/rpc.rs

```diff
@@ -830,35 +830,158 @@ mod tests {
+d570bf3f57c79e5c3309768ee163bf195ab71a608468829a724601e969ec02b66c53b75b59a4fa486aee40054f0ce6614f3fc2ba61b296ce227459dd1d2b72f168398eea5dfc60b4c1c89300348a5f0448bd7cf301f4f15bf9").unwrap();
+            let len = d
+                .decompress_to_buffer(&contract_definition, &mut buffer)
+                .unwrap();
+            buffer.truncate(len);
+            let contract_definition = Bytes::from(buffer);
+
+            {
+                let mut conn = storage.connection().unwrap();
+                let tx = conn.transaction().unwrap();
+
+                crate::state::deploy_contract(
+                    DeployedContract {
+                        address: ContractAddress(
+                            StarkHash::from_hex_str(
+                                "057dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374",
+                            )
+                            .unwrap(),
+                        ),
+                        hash: ContractHash(
+                            StarkHash::from_hex_str(
+                                "050b2148c0d782914e0b12a1a32abe5e398930b7e914f82c65cb7afce0a0ab9b",
+                            )
+                            .unwrap(),
+                        ),
+                        call_data: vec![],
+                    },
+                    &tx,
+                    contract_definition,
+                )
+                .context("Deploy testing contract")
+                .unwrap();
+
+                tx.commit().unwrap();
             }
+
+            let sequencer = sequencer::Client::goerli().unwrap();
+            let (__handle, addr) = run_server(
+                SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, 0)),
+                storage,
+                sequencer,
+            )
+            .unwrap();
+
+            let code = client(addr)
+                .request::<ContractCode>(
+                    "starknet_getCode",
+                    rpc_params!(
+                        "0x057dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374"
+                    ),
+                )
+                .await
+                .unwrap();
+
+            assert_ne!(code.abi.to_string(), "");
+            assert_ne!(code.bytecode, vec![]);
+
+            let code_with_named = client(addr)
+                .request::<ContractCode>(
+                    "starknet_getCode",
+                    by_name([(
+                        "contract_address",
+                        json!("0x057dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374"),
+                    )]),
+                )
+                .await
+                .unwrap();
+
+            assert_eq!(code, code_with_named);
```

Comment:
This is now the previously two different tests for the different parameter encodings.

Reply:
Which I've unified to a single loop now.